tutorials/advanced/make_test/stream_workflows.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2016-2023 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+import yaml
+
+import reframe as rfm
+import reframe.core.builtins as builtins
+from reframe.core.meta import make_test
+
+import stream
+
+
+def load_specs():
+    spec_file = os.getenv('STREAM_SPEC_FILE')
+    if spec_file is None:
+        raise ValueError('no spec file specified')
+
+    with open(spec_file) as fp:
+        try:
+            specs = yaml.safe_load(fp)
+        except yaml.YAMLError as err:
+            raise ValueError(f'could not parse spec file: {err}') from err
+
+    return specs
+
+
+def generate_tests(specs):
+    tests = []
+    for i, spec in enumerate(specs['stream_workflows']):
+        test_body = {}
```

Comment:
Is this necessary?

Reply:
No, it's not ;-) Just a leftover. I will remove it.